pom.xml

```diff
@@ -15,153 +15,224 @@
+    </properties>
 
-    <slf4j.version>1.7.36</slf4j.version>
-    <slf4j-reload4j.version>1.7.36</slf4j-reload4j.version>
-    <reload4j.version>1.2.24</reload4j.version>
-    <scala-parser-combinators.version>1.1.2</scala-parser-combinators.version>
-    <commons-lang.version>2.6</commons-lang.version>
-    <openlineage.version>1.22.0</openlineage.version>
-  </properties>
-
-  <profiles>
-    <profile>
-      <id>release</id>
-      <properties>
-        <gpg.skip>false</gpg.skip>
-      </properties>
-      <build>
-        <plugins>
-          <plugin>
-            <groupId>org.sonatype.plugins</groupId>
-            <artifactId>nexus-staging-maven-plugin</artifactId>
-            <version>${nexus-staging-maven-plugin.version}</version>
-            <extensions>true</extensions>
-            <configuration>
-              <serverId>ossrh</serverId>
-              <nexusUrl>https://oss.sonatype.org/</nexusUrl>
-              <autoReleaseAfterClose>false</autoReleaseAfterClose>
-              <skipRemoteStaging>${nexus.remote.skip}</skipRemoteStaging>
-            </configuration>
-          </plugin>
-          <plugin>
-            <groupId>org.apache.maven.plugins</groupId>
-            <artifactId>maven-gpg-plugin</artifactId>
-            <version>${maven-gpg-plugin.version}</version>
-            <executions>
-              <execution>
-                <id>sign-artifacts</id>
-                <phase>verify</phase>
-                <goals>
-                  <goal>sign</goal>
-                </goals>
-              </execution>
-            </executions>
-            <configuration>
-              <!-- Expect the password to be passed as the gpg.passphrase property
-                    instead of provided over a tty by the gpg-agent process. -->
-              <useAgent>false</useAgent>
-            </configuration>
-          </plugin>
-          <plugin>
-            <groupId>org.apache.maven.plugins</groupId>
-            <artifactId>maven-source-plugin</artifactId>
-            <version>${maven-source-plugin.version}</version>
-            <executions>
-              <execution>
-                <id>attach-sources</id>
-                <goals>
-                  <goal>jar-no-fork</goal>
-                </goals>
-              </execution>
-            </executions>
-          </plugin>
-          <plugin>
-            <groupId>org.apache.maven.plugins</groupId>
-            <artifactId>maven-javadoc-plugin</artifactId>
-            <version>${maven-javadoc-plugin.version}</version>
-            <executions>
-              <execution>
-                <id>attach-javadocs</id>
-                <goals>
-                  <goal>jar</goal>
-                </goals>
-              </execution>
-            </executions>
-          </plugin>
-        </plugins>
-      </build>
-    </profile>
-  </profiles>
+    <profiles>
+        <profile>
+            <id>spark-bt_2.13.14_3.3.3</id>
```

Comment:
Can we name the profiles more descriptively? e.g. `spark-bigtable - scala 2.13 - spark 3.3`

For the artifactIds I'm wondering if we should do something similar. Maybe `spark-bigtable_scala2.12_spark3.3`? Then we could name the profiles the same

I think we can also drop the patch versions (unless you think it would be relevant to keep profiles for multiple patch versions)